torch/_inductor/lowering.py

```diff
@@ -3358,6 +3358,16 @@ def _unsafe_index_put_(self, indices, values, accumulate=False):
 
 
 def index_put_impl_(self, indices, values, accumulate, check):
+    def try_get_name(x):
+        if isinstance(x, ir.TensorBox):
+            x = x.data
+        if isinstance(x, ir.StorageBox):
+            x = x.data
+        return x.get_name() if isinstance(x, ir.Buffer) else None
+
+    if try_get_name(self) in values.get_read_names():
+        return index_put_fallback(self, indices, values, accumulate)
```

Comment:
How about doing just
```suggestion
        values.realize()
```